R/g.sib.det.R

```diff
@@ -26,7 +26,10 @@ g.sib.det = function(M, IMP, I, twd = c(-12, 12),
     time_SPTE_end_hr = as.numeric(format(time_SPTE_end, format = "%H"))
     time_SPTE_start_hr = as.numeric(format(time_SPTE_start, format = "%H"))
     t0 = as.numeric(format(iso8601chartime2POSIX(tmpTIME[1], tz = tz), format = "%H"))
-    t1 = as.numeric(format(iso8601chartime2POSIX(tmpTIME[length(tmpTIME)], tz = tz), format = "%H"))
+    # redefinition of t1 in case window ends some epochs before 12:00:00
+    t1hms = format(iso8601chartime2POSIX(tmpTIME[length(tmpTIME)], tz = tz), format = "%H:%M:%S")
+    t1hms = as.numeric(unlist(strsplit(t1hms, ":", fixed = T)))
+    t1 = ceiling(sum(c(t1hms[1], t1hms[2]/60, t1hms[3]/3600))) 
```

Comment:
I am trying to understanding whether this change affect the functionality of the code.

The main objective of this section is to identify whether there was DST or not in the noon-noon window. We do this by comparing time passed based on number of epochs and time passed based on difference in timestamps.
- If delta time from timestamps > delta time from epochs + 0.1 then clock moved forward.
- If delta time from timestamps < delta time from epochs + 0.1 then clock moved backward.
I am not sure now why I used 0.1.

Can you confirm that this still works and DST is correctly handled?

Reply:
I am now investigating this, and the error in the master branch only appears when the last timestamp is 11:59:55, then the day is considered to be a DST day even if it is not. Also, the approach only adjusts the `SPT_end` time, but not the `SPT_start` time (if sleep onset occurs after 2am, then the start time also needs to be adjusted).

In my approach in this branch, the issue with 11:59:55 is correctly handled, but I have found other errors (it fails when the last timestamp is earlier than 11:59:55), so I need to work more on this.

I will reverse the changes here and try to find a way to solve the original issue without introducing other bugs... I am thinking about getting the timezone specification (`"%z"`) from the ISO8601 format for the first timestamp (e.g., `"+0100"`), and get the same for the spt_start and spt_end timestamps. If `%z` is different between `t0` and `SPT_start `and/or `SPT_end`, then the time would need to be adjusted.

Reply:
Now this simplified option is implemented and it has been tested in different scenarios.